kensu/google/cloud/bigquery/client.py

```diff
@@ -63,11 +69,22 @@ def query(
                         query=query)
 
                     try:
-                        index_select = query.lower().index("select")
-                        query_without_insert = query[index_select:]
-                        logger.debug(f"Query without INSERT TO:{query_without_insert}")
+                        if query.lower().replace(" ","").replace("\n","").startswith("insertinto") :
+                            index_select = query.lower().index("select")
+                            query_without_insert = query[index_select:]
+                            logger.debug(f"Query without INSERT TO:{query_without_insert}")
+                        elif query.lower().replace(" ","").replace("\n","").startswith("merge"):
+                            import re
+                            query_without_insert = re.findall('\(.*\)', query)[0]
```

Comment:
@SammyEK be careful that at least in generic SQL, it's possible to write a MERGE statement without the `(` and `)`, and/or the `WHEN` clauses could be INSIDE the `( ... )`
so in longer term using a proper parser would be safer here.

 see https://stackoverflow.com/questions/46613907/does-apache-spark-sql-support-merge-clause